frontend/src/routes/dashboard/connections/new/+page.svelte

```diff
@@ -1,2 +1,73 @@
+<script lang="ts">
+    /** @type {import('./$types').PageData} */
 
-<sl-button href="/dashboard/connections/new/duckdb">DuckDB</sl-button>
+    import { onMount } from "svelte";
+    import AlertAppError from "$lib/AlertAppError.svelte";
+    import DestinationForm from "$lib/DestinationForm.svelte";
+
+    let activeTab;
+    let tabForms = {};
+    let alertAppError;
+
+    function onTabShow(event) {
+        const connectionType = event.detail.name;
+        const destinationForm = tabForms[connectionType];
+
+        activeTab = connectionType;
+        destinationForm.load(connectionType);
+    }
+
+    onMount(async () => {
+        await Promise.allSettled([
+            customElements.whenDefined("sl-tab-group"),
+            customElements.whenDefined("sl-tab-panel"),
+            customElements.whenDefined("sl-tab"),
+        ]);
+
+        const tabGroup = document.querySelector("sl-tab-group");
+        const activeTabElement = tabGroup.querySelector("sl-tab[active]");
+
+        tabGroup.addEventListener("sl-tab-show", onTabShow);
+
+        if (!activeTab && activeTabElement) {
+            const event = {
+                detail: {
+                    name: activeTabElement.panel,
+                },
+            };
+            onTabShow(event);
+        }
+    });
+</script>
+
+<h2>New Connection</h2>
+
+<AlertAppError bind:this="{alertAppError}" />
```

Comment:
Can you help me understand what ` bind:this="{alertAppError}"` is doing in this file? I'm not totally understanding how the `AlertAppError.svelte` component is supposed to work.

Reply:
This binds the entire component to the name of the parent component. It is later then passed into the component of each form.

Since you would like to have separate routes for each _connection type_, I will likely use a slightly different pattern anyway.